src/osd/PG.cc

```diff
@@ -7251,22 +7237,26 @@ boost::statechart::result PG::RecoveryState::Active::react(const AdvMap& advmap)
   }
 
   bool need_publish = false;
+  pg->_update_calc_stats(); // Too bad this is called again from publish_stats_to_osd()
+  int64_t degraded = pg->info.stats.stats.sum.num_objects_degraded;
+  if (!pg->state_test(PG_STATE_DEGRADED) && degraded) {
+    need_publish = true;
+    pg->state_set(PG_STATE_DEGRADED);
+  } else if (pg->state_test(PG_STATE_DEGRADED) && !degraded) {
+    need_publish = true;
+    pg->state_clear(PG_STATE_DEGRADED);
+  }
+
```

Comment:
Hmm, this doesn't seem like the right place for this (AdvMap).  The num_objects_degraded is not going to change during an osdmap change, but rather during recovery completion or whatever.

Would it make sense to put this in update_calc_stats?